### ISSUE TITLE
MSBuild task for creating Tar files

### ISSUE THREAD

**Issue by stan-sz** [dotnet/msbuild#7944]
With .Net 7 support for [tar apis](https://devblogs.microsoft.com/dotnet/announcing-dotnet-7-preview-4/#added-new-tar-apis), is it feasible to create an equivalent of [Un](https://docs.microsoft.com/en-us/visualstudio/msbuild/unzip-task?view=vs-2022)/[ZipDirectory](https://docs.microsoft.com/en-us/visualstudio/msbuild/zipdirectory-task?view=vs-2022) tasks for handling `.tar` files in the upcoming MSBuild release?

**Comment by danmoseley** [comment]
Cc @carlossanlop fyi

**Comment by danmoseley** [comment]
No it would only work on .NET 7 and later. I believe that would mean .NET SDK 7.0.something and later, targeting any version.

**Comment by rainersigwald** [comment]
We don't yet have any inbox tasks that are .NET (core) only, which might be enough to delay this. But I think it sounds like a reasonable idea.

**Comment by rainersigwald** [comment]
@KalleOlaviNiemitalo Yes, if the `UsingTask` is authored correctly (that is, specifies that the task is .NET-only).

**Comment by baronfel** [comment]
This is a great idea, and we should do it.

_However_, this task would require a .NET 7 Runtime, which brings up an interesting conundrum. That would require MSBuild to update its TFM to net7.0, but for support reasons we _really really_ try to keep components that ship in the .NET SDK to LTS TFMs. We're having a discussion about the mechanics of this later this week, but in general because of SDK and VS support cycles and overlaps (or lack thereof) between the two, we try to limit the amount of STS (short-term support)-only features we use in the product to enable us to fallback to running on supported runtimes when the STS stables go out of support.

That would mean that this feature would either need to be conditioned somehow, or wait until we start targeting net8.0 for MSBuild (which would be something like the 17.8 MSBuild/VS version numbers if memory serves).

**Comment by BenVillalobos** [comment]
In that case, blocked on https://github.com/dotnet/msbuild/pull/7790 and should be conditioned. We can add this to the next spring planning and discuss it then. Conditioning out the task should be easy enough.

**Comment by stan-sz** [comment]
@BenVillalobos can this work be scheduled for a minor release still in 7.0 timeframe (before .NET 8.0)?

**Comment by rainersigwald** [comment]
@stan-sz MSBuild releases on the .NET SDK and Visual Studio cadence, so can add features ~quarterly.

**Comment by stan-sz** [comment]
A kind reminder about this feature request, @benvillalobos and @rainersigwald. Thanks!

**Comment by rainersigwald** [comment]
@stan-sz Are you ok with this only working in `dotnet build`, and not working in Visual Studio/MSBuild.exe?

**Comment by alexrp** [comment]
I would also love to see this. I'm currently doing [this](https://github.com/vezel-dev/zig-toolsets/blob/7d1f1be94a3d34102d92d600138795f6ec7ff9c9/src/Directory.Build.targets#L38-L44), which is... not great, for multiple reasons. (It only working in `dotnet build` would be fine in my case, since these projects are never evaluated/built from anything else, for whatever that's worth.)

**Comment by stan-sz** [comment]
> @stan-sz Are you ok with this only working in `dotnet build`, and not working in Visual Studio/MSBuild.exe?

Yes

**Comment by stan-sz** [comment]
@rainersigwald - a kind reminder so this task makes it to the release. Thanks!

**Comment by just-ero** [comment]
This feature would be very helpful for automating use of the [WASI SDK](https://github.com/WebAssembly/wasi-sdk), which is archived using the Tarball format.
